cmd/cogito/main_test.go

```diff
@@ -134,6 +135,46 @@ func TestRunPutSuccessIntegration(t *testing.T) {
 		`level=INFO msg="state posted successfully to chat" name=cogito.put name=gChat state=error`))
 }
 
+func TestRunPutGhAppSuccessIntegration(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration test (reason: -short)")
+	}
+
+	gitHubCfg := testhelp.GitHubSecretsOrFail(t)
+	ghAppInstallationID, err := strconv.ParseInt(gitHubCfg.GhAppInstallationID, 10, 32)
+	assert.NilError(t, err)
+
+	stdin := bytes.NewReader(testhelp.ToJSON(t, cogito.PutRequest{
+		Source: cogito.Source{
+			Owner: gitHubCfg.Owner,
+			Repo:  gitHubCfg.Repo,
+			GitHubApp: github.GitHubApp{
+				ClientId:       gitHubCfg.GhAppClientID,
+				InstallationId: int(ghAppInstallationID),
+				PrivateKey:     gitHubCfg.GhAppPrivateKey,
+			},
+			LogLevel: "debug",
+		},
+		Params: cogito.PutParams{State: cogito.StateError},
+	}))
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+	inputDir := testhelp.MakeGitRepoFromTestdata(t, "../../cogito/testdata/one-repo/a-repo",
+		testhelp.HttpsRemote(github.GhDefaultHostname, gitHubCfg.Owner, gitHubCfg.Repo), gitHubCfg.SHA,
+		"ref: refs/heads/a-branch-FIXME")
+	t.Setenv("BUILD_JOB_NAME", "TestRunPutGhAppSuccessIntegration")
+	t.Setenv("ATC_EXTERNAL_URL", "https://cogito.invalid")
```

Comment:
I don't think the value actually matters but did you chose the `invalid` TLD on purpose?
I'm asking because it is a reserved one with special behaviour defined by [RFC6761 section 6.4](https://www.rfc-editor.org/rfc/rfc6761.html#section-6.4). If there is no real intent to have an NXDOMAIN response, maybe using `example` would be better (see [RFC6761 section 6.5](https://www.rfc-editor.org/rfc/rfc6761.html#section-6.5).

Reply:
I must say: I don't have any smart answer except... I copy/pasted it from existing tests...

https://github.com/Pix4D/cogito/blob/fa53eabfaba7dff0d6de9fc94306e3c7c5f81e57/cmd/cogito/main_test.go#L123

But, you are correct, I remember the same discussion in some other project (skyguide maybe?)... Anyway... I will not address this comment in this PR, but I can prepare a separate PR to fix it everywhere...

@odormond ok?

Reply:
Fine for me. Definitely a very minor point.